### ISSUE TITLE
Allow Subpixel ScrollContainer H and V Scroll Values

### ISSUE THREAD

**Issue by iiMidknightii** [godotengine/godot-proposals#10544]
### Describe the project you are working on

A card game with sometimes dense menus, requiring auto-scrolling text and panels.

### Describe the problem or limitation you are having in your project

I created a custom ScrollContainer script that automatically scrolls its contents either vertically or horizontally like a ticker tape or new crawl.  This allows me to display large amounts of horizontal text in a small horizontal space.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Currently, ScrollContainer only allows integer values for `h_scroll` and `v_scroll`.  This makes it difficult to scroll smoothly in `_process` as the scroll speed either needs to be large enough to always be >1 when multiplied by `delta`, or it won't scroll at all.  This leads to scroll speeds that are not very configurable and have quite a high minimum.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`h_scroll` and `v_scroll` are actually just wrapper methods to set the value of the underlying scroll bars.  That value is internally a float, and it doesn't look like it was rounded, either.  This means that the members of ScrollContainer are already doing a narrowing conversion in the get methods, and expanding integers to floats in the set methods.  I propose the values are changed to floats in ScrollContainer to match.  If necessary, I could also add a `round_to_nearest_pixel` or similar member to force the values back to nearest integer (maybe by setting the internal scrollbars' rounded flag).

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think the exact scrolling behavior I wanted could be achieved with lower rate timers, but it wouldn't be smooth and any other custom ScrollContainers would also have to make custom work-arounds to get smooth scrolling over time.

### Is there a reason why this should be core and not an add-on in the asset library?

The types are inherent to the built-in ScrollContainer.

**Comment by Mickeon** [comment]
I do agree it _looks_ nonsensical. 

But, for your use case, could you use a "wrapper" float value yourself? Or, alternatively, interpolate `h_scroll`/`h_scroll` with a Tween?

**Comment by iiMidknightii** [comment]
> I do agree it _looks_ nonsensical.
> 
> But, for your use case, could you use a "wrapper" float value yourself? Or, alternatively, interpolate `h_scroll`/`h_scroll` with a Tween?

I guess that's also a possibility, but at slow speeds it's not very smooth, especially on lower resolution games.

**Comment by RedMser** [comment]
> at slow speeds it's not very smooth, especially on lower resolution games

As workaround, try `scroll_container.get_v_scroll_bar().value = my_float` instead, maybe it works? I have not tried this myself, but looking at the code it should be the same as setting `v_scroll` (besides a missing call to `_cancel_drag()`, which might cause issues dragging while animating the value).
If this isn't smoother, then more would need to be done to properly support floats in scroll container anyway.

**Comment by Mickeon** [comment]
They are saying that interpolation would be "smooth", but the actual scroll would still snap to integer coordinates, which is noticeable when zooming in the **ScrollContainer** in some way.

**Comment by iiMidknightii** [comment]
I think this can be closed.  I ended up using the "wrapper" float, and if the actual graphics still snap to the nearest pixel then integer makes the most sense.

**Comment by Mickeon** [comment]
I'd still champion for this, even if it cannot be done now. Should be done next time breaking compatibility is feasible.

**Comment by PostPollux** [comment]
@Mickeon 

Yes it is true that the proposed solution still results in a scroll that snaps to pixels. But that can easily be fixed as well. For details see my comment I made on the pull request linked to this proposal.
